### ISSUE TITLE
Panic with `AccessError` when logging in a `drop`

### ISSUE THREAD

**Issue by drmingdrmer** [datafuselabs/databend#12513]
### Summary:
When `log::trace!()` is called from in a `Drop`, `minitrace` tries to access destructed thread-local variable `LOCAL_SPAN_STACK`, which cause the panic: `cannot access a Thread Local Storage value during or after destruction: AccessError`.

### Cause:

It looks like the `log::trace!()` in a `Drop` is called when a thread is destructing. But during thread destruction, the `minitrace` thread local variable `LOCAL_SPAN_STACK` is already destructed too.

### Solution:

A possible solution to this issue would be replacing `LOCAL_SPAN_STACK.with()` with `LOCAL_SPAN_STACK.try_with()` to prevent an `Event` from accessing destructed `LOCAL_SPAN_STACK`, in the following codes:

https://github.com/tikv/minitrace-rust/blob/39d8c9683d96f994a7507abdc933d6bc363256d2/minitrace/src/event.rs#L51-L63
```rust
    pub fn add_to_local_parent<I, F>(name: &'static str, properties: F)
    where
        I: IntoIterator<Item = (Cow<'static, str>, Cow<'static, str>)>,
        F: FnOnce() -> I,
    {
        #[cfg(feature = "enable")]
        {
            let stack = LOCAL_SPAN_STACK.with(Rc::clone);
            let mut stack = stack.borrow_mut();
            stack.add_event(name, properties);
        }
    }
}
```

### Failed CI that encounter this panic:
https://github.com/datafuselabs/databend/actions/runs/5912042028/job/16035066707?pr=12508

### Backtrace:
```
2023-08-19T15:05:37.845640Z ERROR common_tracing::panic_hook: src/common/tracing/src/panic_hook.rs:42 panicked at 'cannot access a Thread Local Storage value during or after destruction: AccessError', /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/std/src/thread/local.rs:246:26 backtrace=Backtrace [
{ fn: "common_tracing::panic_hook::log_panic", file: "/workspace/src/common/tracing/src/panic_hook.rs", line: 35 }, 
{ fn: "common_tracing::panic_hook::set_panic_hook::{{closure}}", file: "/workspace/src/common/tracing/src/panic_hook.rs", line: 30 }, 
{ fn: "call<(&core::panic::panic_info::PanicInfo), (dyn core::ops::function::Fn<(&core::panic::panic_info::PanicInfo), Output=()> + core::marker::Send + core::marker::Sync), alloc::alloc::Global>", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/alloc/src/boxed.rs", line: 1999 }, 
{ fn: "rust_panic_with_hook", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/std/src/panicking.rs", line: 709 }, 
{ fn: "{closure#0}", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/std/src/panicking.rs", line: 597 }, 
{ fn: "__rust_end_short_backtrace<std::panicking::begin_panic_handler::{closure_env#0}, !>", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/std/src/sys_common/backtrace.rs", line: 151 }, 
{ fn: "begin_panic_handler", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/std/src/panicking.rs", line: 593 }, 
{ fn: "panic_fmt", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/panicking.rs", line: 67 }, 
{ fn: "unwrap_failed", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/result.rs", line: 1651 }, 
{ fn: "core::result::Result<T,E>::expect", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/result.rs", line: 1033 }, 
{ fn: "std::thread::local::LocalKey<T>::with", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/std/src/thread/local.rs", line: 246 }, 
{ fn: "minitrace::event::Event::add_to_local_parent", file: "/opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/minitrace-0.5.0/src/event.rs", line: 58 }, 
{ fn: "<common_tracing::minitrace::MinitraceLogger as log::Log>::log", file: "/workspace/src/common/tracing/src/minitrace.rs", line: 279 }, 
{ fn: "<alloc::boxed::Box<T> as log::Log>::log", file: "/opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/log-0.4.19/src/lib.rs", line: 1192 }, 
{ fn: "<fern::log_impl::Output as log::Log>::log", file: "/opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/fern-0.6.2/src/log_impl.rs", line: 355 }, 
{ fn: "fern::log_impl::Dispatch::finish_logging", file: "/opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/fern-0.6.2/src/log_impl.rs", line: 464 }, 
{ fn: "<fern::log_impl::Dispatch as log::Log>::log", file: "/opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/fern-0.6.2/src/log_impl.rs", line: 448 }, 
{ fn: "<fern::log_impl::Output as log::Log>::log", file: "/opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/fern-0.6.2/src/log_impl.rs", line: 353 }, 
{ fn: "fern::log_impl::Dispatch::finish_logging", file: "/opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/fern-0.6.2/src/log_impl.rs", line: 464 }, 
{ fn: "<fern::log_impl::Dispatch as log::Log>::log", file: "/opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/fern-0.6.2/src/log_impl.rs", line: 448 }, 
{ fn: "<fern::log_impl::Output as log::Log>::log", file: "/opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/fern-0.6.2/src/log_impl.rs", line: 353 }, 
{ fn: "fern::log_impl::Dispatch::finish_logging", file: "/opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/fern-0.6.2/src/log_impl.rs", line: 464 }, 
{ fn: "<fern::log_impl::Dispatch as log::Log>::log", file: "/opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/fern-0.6.2/src/log_impl.rs", line: 448 }, 
{ fn: "<fern::log_impl::Output as log::Log>::log", file: "/opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/fern-0.6.2/src/log_impl.rs", line: 353 }, 
{ fn: "fern::log_impl::Dispatch::finish_logging", file: "/opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/fern-0.6.2/src/log_impl.rs", line: 464 }, 
{ fn: "<fern::log_impl::Dispatch as log::Log>::log", file: "/opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/fern-0.6.2/src/log_impl.rs", line: 448 }, 
{ fn: "log::__private_api_log", file: "/opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/log-0.4.19/src/lib.rs", line: 1503 }, 
{ fn: "sled::pagecache::iobuf::IoBufs::write_to_log::{{closure}}", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/pagecache/iobuf.rs", line: 875 }, 
{ fn: "sled::ebr::Guard::defer_unchecked::{{closure}}", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/mod.rs", line: 47 }, 
{ fn: "sled::ebr::deferred::call_raw", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/deferred.rs", line: 32 }, 
{ fn: "sled::ebr::deferred::Deferred::call", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/deferred.rs", line: 81 }, 
{ fn: "<sled::ebr::internal::Bag as core::ops::drop::Drop>::drop", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/internal.rs", line: 200 }, 
{ fn: "core::ptr::drop_in_place<sled::ebr::internal::Bag>", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/ptr/mod.rs", line: 497 }, 
{ fn: "core::ptr::drop_in_place<sled::ebr::internal::SealedBag>", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/ptr/mod.rs", line: 497 }, 
{ fn: "core::mem::drop", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/mem/mod.rs", line: 987 }, 
{ fn: "sled::ebr::internal::Global::collect", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/internal.rs", line: 295 }, 
{ fn: "sled::ebr::internal::Local::pin", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/internal.rs", line: 529 }, 
{ fn: "sled::ebr::collector::LocalHandle::pin", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/collector.rs", line: 74 }, 
{ fn: "core::ops::function::FnMut::call_mut", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/ops/function.rs", line: 166 }, 
{ fn: "sled::ebr::collector::with_handle::{{closure}}", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/collector.rs", line: 29 }, 
{ fn: "core::result::Result<T,E>::unwrap_or_else", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/result.rs", line: 1428 }, 
{ fn: "sled::ebr::collector::with_handle", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/collector.rs", line: 29 }, 
{ fn: "sled::ebr::collector::pin", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/collector.rs", line: 21 }, 
{ fn: "<sled::stack::Node<T> as core::ops::drop::Drop>::drop", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/stack.rs", line: 24 }, 
{ fn: "core::ptr::drop_in_place<sled::stack::Node<sled::pagecache::segment::SegmentOp>>", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/ptr/mod.rs", line: 497 }, 
{ fn: "core::ptr::drop_in_place<alloc::boxed::Box<sled::stack::Node<sled::pagecache::segment::SegmentOp>>>", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/ptr/mod.rs", line: 497 }, 
{ fn: "core::mem::drop", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/mem/mod.rs", line: 987 }, 
{ fn: "<T as sled::ebr::atomic::Pointable>::drop", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/atomic.rs", line: 192 }, 
{ fn: "<sled::ebr::atomic::Owned<T> as core::ops::drop::Drop>::drop", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/atomic.rs", line: 605 }, 
{ fn: "core::ptr::drop_in_place<sled::ebr::atomic::Owned<sled::stack::Node<sled::pagecache::segment::SegmentOp>>>", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/ptr/mod.rs", line: 497 }, 
{ fn: "core::mem::drop", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/mem/mod.rs", line: 987 }, 
{ fn: "sled::ebr::Guard::defer_unchecked::{{closure}}", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/mod.rs", line: 47 }, 
{ fn: "sled::ebr::deferred::call_raw", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/deferred.rs", line: 32 }, 
{ fn: "sled::ebr::deferred::Deferred::call", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/deferred.rs", line: 81 }, 
{ fn: "<sled::ebr::internal::Bag as core::ops::drop::Drop>::drop", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/internal.rs", line: 200 }, 
{ fn: "core::ptr::drop_in_place<sled::ebr::internal::Bag>", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/ptr/mod.rs", line: 497 }, 
{ fn: "core::ptr::drop_in_place<sled::ebr::internal::SealedBag>", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/ptr/mod.rs", line: 497 }, 
{ fn: "core::mem::drop", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/mem/mod.rs", line: 987 }, 
{ fn: "sled::ebr::internal::Global::collect", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/internal.rs", line: 295 }, 
{ fn: "sled::ebr::internal::Local::pin", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/internal.rs", line: 529 }, 
{ fn: "sled::ebr::internal::Local::finalize", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/internal.rs", line: 578 }, 
{ fn: "sled::ebr::internal::Local::release_handle", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/internal.rs", line: 562 }, 
{ fn: "<sled::ebr::collector::LocalHandle as core::ops::drop::Drop>::drop", file: "/opt/rust/cargo/git/checkouts/sled-95f75fbdab2965fb-shallow/43fa725/src/ebr/collector.rs", line: 82 }, 
{ fn: "core::ptr::drop_in_place<sled::ebr::collector::LocalHandle>", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/ptr/mod.rs", line: 497 }, 
{ fn: "core::ptr::drop_in_place<core::option::Option<sled::ebr::collector::LocalHandle>>", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/ptr/mod.rs", line: 497 }, 
{ fn: "std::sys::common::thread_local::fast_local::destroy_value::{{closure}}", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/mem/mod.rs", line: 987 }, 
{ fn: "core::ops::function::FnOnce::call_once", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/ops/function.rs", line: 250 }, 
{ fn: "<core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/panic/unwind_safe.rs", line: 271 }, 
{ fn: "std::panicking::try::do_call", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/std/src/panicking.rs", line: 500 }, 
{ fn: "__rust_try" }, 
{ fn: "std::panicking::try", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/std/src/panicking.rs", line: 464 }, 
{ fn: "std::sys::common::thread_local::fast_local::destroy_value", file: "/rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/std/src/panic.rs", line: 142 }, 
{ fn: "__call_tls_dtors" }, 
{ fn: "start_thread" }, 
{ fn: "clone" }] 

```

**Comment by drmingdrmer** [comment]
@ariesdevil  @andylokandy

**Comment by drmingdrmer** [comment]
Apply this minitrace patch to databend:
- https://github.com/tikv/minitrace-rust/pull/161

**Comment by zhongzc** [comment]
This is fixed in minitrace@v0.5.1

**Comment by drmingdrmer** [comment]
> This is fixed in minitrace@v0.5.1

Thanks for informing that!
